network/src/network.rs

```diff
@@ -14,14 +14,86 @@
 
 //! SSV network
 
-use crate::{p2p::transport, Result};
+use crate::{behaviour::Behaviour, p2p::transport, Result};
+use anyhow::anyhow;
+use libp2p::{
+    bandwidth::BandwidthSinks,
+    identify::{Behaviour as Identify, Config as IdentifyConfig},
+    swarm::{Swarm, SwarmBuilder},
+    Multiaddr, PeerId,
+};
+use log::{info, trace};
+use std::{net::SocketAddrV4, sync::Arc};
 
 /// SSV network
-pub struct Network;
+pub struct Network {
+    swarm: Swarm<Behaviour>,
+    /// The bandwidth logger for the underlying libp2p transport.
+    pub bandwidth: Arc<BandwidthSinks>,
+}
 
 impl Network {
+    /// New network
     pub fn new() -> Result<Self> {
-        // let _ = transport::build()?;
-        Ok(Network)
+        // FIXME: Load keypair from config (issue #11)
+        let local_keypair = libp2p::core::identity::Keypair::generate_ed25519();
+
+        // NOTE: Testing for now (issue #21)
+        //
+        // TODO:
+        // - gossipsub (issue #25)
+        // - discovery (issue #9)
+        // - peer manager (issue #22)
+        let behaviour = Behaviour {
+            // FIXME: testing (issue #23)
+            identify: Identify::new(IdentifyConfig::new(
+                "ssv/0.0.1".into(),
```

Comment:
```suggestion
                "ssv-rs/0.0.1".into(),
```

there will be the day when you can be creative and give this project a name :)